src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/InstrumentAdvice.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace System.Diagnostics.Metrics
+{
+    /// <summary>
+    /// Contains configuration settings advised to be used by metrics consumers when recording measurements for a given <see cref="Instrument{T}"/>.
+    /// </summary>
+    /// <typeparam name="T">Instrument value type.</typeparam>
+    public sealed class InstrumentAdvice<T> where T : struct
```

Comment:
>public sealed class InstrumentAdvice<T> where T : struct [](http://example.com/codeflow?start=4&length=56)

I think we need a parameter-less constructor here and inside it we need to call `Instrument.ValidateTypeParameter<T>()`

[https://github.com/dotnet/runtime/blob/29f3110fc971c8cc48f18956d347e58c54bd0639/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Instrument.cs#L141C30-L141C38](https://github.com/dotnet/runtime/blob/29f3110fc971c8cc48f18956d347e58c54bd0639/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Instrument.cs#L141C30-L141C38)

Reply:
Updated

Reply:
do we need to add the constructor to the ref file too `src\libraries\System.Diagnostics.DiagnosticSource\ref\System.Diagnostics.DiagnosticSourceActivity.cs`?

Reply:
It doesn't complain without it so maybe it is an implied thing? I went ahead and added it though.

Reply:
Right. the compiler automatically generates the parameter-less constructor. So, it implicitly existed in the ref too. It is good to explicitly list it as we add some code there.